src/Compilers/Core/Portable/StrongName/StrongNameProvider.cs

```diff
@@ -16,11 +26,21 @@ protected StrongNameProvider()
         public abstract override bool Equals(object other);
         public abstract override int GetHashCode();
 
+        internal abstract SigningCapability Capability { get; }
+
         /// <exception cref="IOException"></exception>
         internal abstract Stream CreateInputStream();
 
         internal abstract StrongNameKeys CreateKeys(string keyFilePath, string keyContainerName, CommonMessageProvider messageProvider);
 
-        internal abstract void SignAssembly(StrongNameKeys keys, Stream inputStream, Stream outputStream);
+        internal virtual void SignStream(StrongNameKeys keys, Stream inputStream, Stream outputStream)
+        {
+            throw new InvalidOperationException();
+        }
+
+        internal virtual void SignPeBuilder(ExtendedPEBuilder peBuilder, BlobBuilder peBlob, RSAParameters privkey)
+        {
+            throw new InvalidOperationException();
+        }
```

Comment:
If there are only a few derived types, consider making these two methods `abstract`, to ensure the methods are implemented.

Reply:
That's how I had it, but @jaredpar said to move the method to the base type.